core/storage/btree.rs

```diff
@@ -4484,7 +4614,7 @@ impl BTreeCursor {
                             self.overflow_state = OverflowState::Start;
                             return Err(LimboError::Corrupt("Invalid overflow page number".into()));
                         }
-                        let (page, c) = self.read_page(next as i64)?;
+                        let (page, c) = return_if_io!(self.read_page(next as i64));
```

Comment:
How does this work? If we yield on IO here, then `overflow_state` won't be changed. If we don't yield on IO here, then is `c` ever `Some`?